### PR TITLE
fix: Fix warnings during documentation building

### PR DESCRIPTION
**Pull request by paulinebanye**
### Description
This pull request attempts to fix the warnings encountered during the `make html` process -  issue #1202

---

### **Environment**

- Model: Dell Latitude E7440
- RAM: 12GB 
- OS: Windows 10 pro
- Python Version: Python 3.12
- Terminal: Ubuntu 22.04 LTS

---

### Steps to reproduce:
1. Clone the skore repository
2. Create and activate a virtual environment using the Ubuntu terminal
3. Run the _make install-skore_ command.
4. Cd into the sphinx directory
5. Run `make html`.
---

### Warnings:
Listed below are the warnings encountered during the process.

#### 1. Title underline too short
- **Cause**\
    These warnings occurred because the title and underline had different lengths. 

```python skore/sphinx/auto_examples/use_cases/plot_employee_salaries.rst:14417: Tree-based model =============== [docutils]
skore/sphinx/auto_examples/use_cases/plot_employee_salaries.rst:14417: Title underline too short. Tree-based model =============== [docutils]
skore/sphinx/auto_examples/use_cases/plot_employee_salaries.rst:15060: Title underline too short. Linear model ========= [docutils]
skore/sphinx/auto_examples/use_cases/plot_employee_salaries.rst:15060: Title underline too short. Linear model ========= [docutils]
skore/sphinx/auto_examples/use_cases/plot_feature_importance.rst:12457: Title underline too short. Splitting the data ----------------- [docutils]
skore/sphinx/auto_examples/use_cases/plot_feature_importance.rst:12457: Title underline too short. Splitting the data ----------------- [docutils]
```

- **Steps taken:**
   - Identified the python files that generate the rst files (in examples/use_cases).
  - Modified the code to ensure that the underlines matched the length of the titles. 

--- 

#### 2. Duplicate explicit target name
- **Cause**\
This warning occurred because two links had the same text as their visible label.

```python
skore/sphinx/auto_examples/use_cases/plot_feature_importance.rst:25: WARNING: Duplicate explicit target name: "scikit-learn's user guide". [docutils]
```

- **Steps taken:**
   - Identified the python file that generates the rst files (examples/use_cases/plot_feature_importance.py).
   - Changed the target name for one of the occurrences from "scikit-learn's user guide" to "scikit-learn's documentation".

---

### 3. Duplicate object description
**Cause**\
These warnings indicate that object descriptions are duplicated in the documentation, and suggest using `:no-index:` for one of them. This duplication is happening because the object is already described in the .rst files (like comparison_report.rst), and Sphinx is also automatically adding it, so it ends up showing up twice.

```python
# ComparisonReport`:
skore/skore/src/skore/sklearn/_comparison/report.py:docstring of skore.sklearn._comparison.report.ComparisonReport.cache_predictions:1: WARNING: duplicate object description of skore.ComparisonReport.cache_predictions, other instance in reference/api/skore.ComparisonReport, use :no-index: for one of them
skore/skore/src/skore/sklearn/_comparison/report.py:docstring of skore.sklearn._comparison.report.ComparisonReport.clear_cache:1: WARNING: duplicate object description of skore.ComparisonReport.clear_cache, other instance in reference/api/skore.ComparisonReport, use :no-index: for one of them
skore/skore/src/skore/sklearn/_comparison/report.py:docstring of skore.sklearn._comparison.report.ComparisonReport.get_predictions:1: WARNING: duplicate object description of skore.ComparisonReport.get_predictions, other instance in reference/api/skore.ComparisonReport, use :no-index: for one of them
skore/skore/src/skore/sklearn/_base.py:docstring of skore.sklearn._base._HelpMixin.help:1: WARNING: duplicate object description of skore.ComparisonReport.help, other instance in reference/api/skore.ComparisonReport, use :no-index: for one of them
skore/skore/src/skore/sklearn/_comparison/metrics_accessor.py:docstring of skore.sklearn._comparison.metrics_accessor._MetricsAccessor:1: WARNING: duplicate object description of skore.ComparisonReport.metrics, other instance in reference/api/skore.ComparisonReport, use :no-index: for one of them

# CrossValidationReport
skore/skore/src/skore/sklearn/_cross_validation/report.py:docstring of skore.sklearn._cross_validation.report.CrossValidationReport.cache_predictions:1: WARNING: duplicate object description of skore.CrossValidationReport.cache_predictions, other instance in reference/api/skore.CrossValidationReport, use :no-index: for one of them
skore/skore/src/skore/sklearn/_cross_validation/report.py:docstring of skore.sklearn._cross_validation.report.CrossValidationReport.clear_cache:1: WARNING: duplicate object description of skore.CrossValidationReport.clear_cache, other instance in reference/api/skore.CrossValidationReport, use :no-index: for one of them
skore/skore/src/skore/sklearn/_cross_validation/report.py:docstring of skore.sklearn._cross_validation.report.CrossValidationReport.get_predictions:1: WARNING: duplicate object description of skore.CrossValidationReport.get_predictions, other instance in reference/api/skore.CrossValidationReport, use :no-index: for one of them
skore/skore/src/skore/sklearn/_base.py:docstring of skore.sklearn._base._HelpMixin.help:1: WARNING: duplicate object description of skore.CrossValidationReport.help, other instance in reference/api/skore.CrossValidationReport, use :no-index: for one of them
skore/skore/src/skore/sklearn/_cross_validation/metrics_accessor.py:docstring of skore.sklearn._cross_validation.metrics_accessor._MetricsAccessor:1: WARNING: duplicate object description of skore.CrossValidationReport.metrics, other instance in reference/api/skore.CrossValidationReport, use :no-index: for one of them.

# EstimatorReport
skore/skore/src/skore/sklearn/_estimator/report.py:docstring of skore.sklearn._estimator.report.EstimatorReport.cache_predictions:1: WARNING: duplicate object description of skore.EstimatorReport.cache_predictions, other instance in reference/api/skore.EstimatorReport, use :no-index: for one of them
skore/skore/src/skore/sklearn/_estimator/report.py:docstring of skore.sklearn._estimator.report.EstimatorReport.clear_cache:1: WARNING: duplicate object description of skore.EstimatorReport.clear_cache, other instance in reference/api/skore.EstimatorReport, use :no-index: for one of them
skore/skore/src/skore/sklearn/_estimator/feature_importance_accessor.py:docstring of skore.sklearn._estimator.feature_importance_accessor._FeatureImportanceAccessor:1: WARNING: duplicate object description of skore.EstimatorReport.feature_importance, other instance in reference/api/skore.EstimatorReport, use :no-index: for one of them
skore/skore/src/skore/sklearn/_estimator/report.py:docstring of skore.sklearn._estimator.report.EstimatorReport.get_predictions:1: WARNING: duplicate object description of skore.EstimatorReport.get_predictions, other instance in reference/api/skore.EstimatorReport, use :no-index: for one of them
skore/skore/src/skore/sklearn/_base.py:docstring of skore.sklearn._base._HelpMixin.help:1: WARNING: duplicate object description of skore.EstimatorReport.help, other instance in reference/api/skore.EstimatorReport, use :no-index: for one of them
skore/skore/src/skore/sklearn/_estimator/metrics_accessor.py:docstring of skore.sklearn._estimator.metrics_accessor._MetricsAccessor:1: WARNING: duplicate object description of skore.EstimatorReport.metrics, other instance in reference/api/skore.EstimatorReport, use :no-index: for one of them

# Project
skore/skore/src/skore/project/project.py:docstring of skore.project.project.Project.get:1: WARNING: duplicate object description of skore.Project.get, other instance in reference/api/skore.Project, use :no-index: for one of them
skore/skore/src/skore/project/project.py:docstring of skore.project.project.Project.put:1: WARNING: duplicate object description of skore.Project.put, other instance in reference/api/skore.Project, use :no-index: for one of them
```
**Fix:**\
_Not fixed yet._

**Attempts**
- Disabled backreferences for the affected methods by adding `:no-index:` or `:noindex:` to all the affected python files. ❌
   ```python
   def get(self, ...) -> Any:
    """:no-index:"""
   ```
- Modified the rst files to include :no-index:.❌
   ```python
   .. autoclass:: skore.Project
      :members:
      :no-index: 
   ```
- Currently I'm combining editing the conf.py file and excluding documenting the methods explicitly from the rst riles:-
   - Exclude specific methods explicitly from rst and adjust the sphinx_gallery_conf backreferences_dir option (which generates references from `reference/api`) to None.
  ```python
   #conf.py
   sphinx_gallery_conf = {
      "backreferences_dir": None,
   }

  project.rst
  .. autosummary::
      :toctree: api/
      :template: base.rst

    Project # removed manually specifying methods to avoid duplicates.
   ```
- Include autodoc options to exclude specific methods in conf.py
   ```python
   autodoc_default_options = {
       'exclude-members': 'put, get', 
    }
   ```
- Change the autosummary option from True to False in conf.py
  
   ```python
   autosummary_generate = False
   ```

### 4. Configuration cache
**Cause**\
This warning indicates that a configuration value cannot be cached because it contains an unpickable object (like a function, class, or module).

```python
WARNING: cannot cache unpickable configuration value: 'sphinx_gallery_conf' (because it contains a function, class, or module object) [config.cache]
```

**Fix:**\
_Not fixed yet._

**Attempts**
_Not attempted yet._

cc @sylvaincom

### CODE REVIEW

**Comment by paulinebanye** [comment]
cc @sylvaincom

**Comment by paulinebanye** [comment]
> @paulinebanye Could you make separate PR for each case. My feeling here is that some of them are really quick win without question (like the underline). Modifying the config will take a bit more time to get around.

That's actually a really good idea @glemaitre. Thanks 👍🏻.

Let me make PRs for the different fixes.
